Assignment2-Ancestors/Q1.py

```diff
@@ -0,0 +1,44 @@
+from Tree import BinaryTree
+
+def find_ancestors(tree, key):
+    """
+    Find ancestors of the first node with the given value while preorder traversal and save the Nodes in a list that return.
+    :param tree: Tree
+    :param key: any
+    :return: list(Nodes)
+    """
+    node = tree.find(key)
+    if node:

```

Comment:
It can be a lot of work to think of good test cases to catch all reasonable mistakes. I have not directly used for example https://coverage.readthedocs.io/ but I've worked on code bases where something like it had been integrated into workflow to point out test coverage at code review time. That was useful.

I don't know if it would have helped to point out that some code path had no test coverage in this case?